app/lib/otis/jira_client.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require "jira-ruby"
+
+module Otis
+  class JiraClient
+    # Fake JIRA::Client used when Otis.config.jira.readonly is set.
+    # It just responds to most methods by returning itself.
+    class NullJiraObj
```

Comment:
I'd be inclined to move this into the tests and allow it to be injected into `Otis::JiraClient` at constructor time:

```ruby
def initialize(client = connect)
  @client = client
end
```

and then in tests

```ruby
OtisJiraClient.new(client = NullJiraObj.new)
```

Reply:
Yes, I like that approach better.

Reply:
Except we also want a null object in development, in case we accidentally use a real Jira ticket number when running through registration scenarios. What if we put an explicit factory in Otis::JiraClient that explicitly uses Rails.env.production? to determine what to return? And dispense with the "readonly" config?

Reply:
Yes, I think that approach makes sense.